test/Faker/Provider/ImageTest.php

```diff
@@ -45,6 +45,23 @@ public function testUrlWithDimensionsAndBadCategory()
 
     public function testDownloadWithDefaults()
     {
+        $url = "http://www.lorempixel.com/";
+        $curlPing = curl_init($url);
+        curl_setopt($curlPing, CURLOPT_TIMEOUT, 5);
+        curl_setopt($curlPing, CURLOPT_CONNECTTIMEOUT, 5);
+        curl_setopt($curlPing, CURLOPT_RETURNTRANSFER, true);
+        $data = curl_exec($curlPing);
+        $httpCode = curl_getinfo($curlPing, CURLINFO_HTTP_CODE);
+        curl_close($curlPing);
+
+        /**
+         *  If HTTP response is something else than 200-299, skip the test
+         */
+        if ($httpCode < 200 | $httpCode > 300)
+        {
```

Comment:
I'll just clean up the rest of the PR. Do we need squashing the commits or is this OK on it's own?